packages/rum-core/src/domain/lifeCycle.ts

```diff
@@ -10,14 +10,22 @@ import type { AutoAction } from './action/actionCollection'
 import type { ViewEvent, ViewCreatedEvent, ViewEndedEvent } from './view/trackViews'
 
 export const enum LifeCycleEventType {
+  // BEFORE_XXX events have been introduced to provide hooks just before significant actions or transitions in the application lifecycle.
+  // The following three events allow for performing tasks, such as sending last events, since the SDK contexts (viewContext, etc.) are still active.
+  BEFORE_VIEW_ENDED,
+  BEFORE_SESSION_EXPIRED,
+  BEFORE_PAGE_EXITED,
+  // The BEFORE_VIEW_CREATED event has been introduced to provide a hook just before a view is created,
+  // ensuring that the context related to the view (viewContexts, urlContexts) exists at VIEW_CREATED.
+  BEFORE_VIEW_CREATED,
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** Given that most of the listeners should not be related to context management, what about:
- using non-suffixed "XXX starts" / "XXX ends" events for listeners that are not managing contexts
- using suffixed "before XXX starts" / "after XXX ends" events for listeners that are managing contexts  

We would then have: contexts should be created before XXX starts and closed after XXX ends.

Wdyt?

Reply:
Hum interesting. I'll give it a try

Reply:
Let me know what you think ?